test/testchain/chain.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package testchain
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/vechain/thor/v2/bft"
+	"github.com/vechain/thor/v2/block"
+	"github.com/vechain/thor/v2/chain"
+	"github.com/vechain/thor/v2/genesis"
+	"github.com/vechain/thor/v2/muxdb"
+	"github.com/vechain/thor/v2/packer"
+	"github.com/vechain/thor/v2/state"
+	"github.com/vechain/thor/v2/thor"
+	"github.com/vechain/thor/v2/tx"
+)
+
+// Chain represents the blockchain structure.
+// It includes database (db), genesis information (genesis), consensus engine (engine),
+// repository for blocks and state (repo), state manager (stater), and the genesis block (genesisBlock).
+type Chain struct {
```

Comment:
Maybe we can name this TestChain not to confuse it with 

```golang
type Chain struct {
	repo     *Repository
	headID   thor.Bytes32
	lazyInit func() (*muxdb.Trie, error)
}
```

Reply:
I originally had named it TestChain. However the imports would now become `testchain.TestChain` which is somewhat anti pattern.
Given that there is no compatibility between the two Chains I'm more inclined to use this current approach. Or perhaps there is another name we can use ?